sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -1032,10 +1032,12 @@ void coap_block_finish(coap_block_slicer_t *slicer, uint16_t option)
     uint8_t *pos = slicer->opt + 1;
     uint16_t delta = _decode_value(*slicer->opt >> 4, &pos, slicer->opt + 3);
 
-    uint32_t blkopt = _slicer2blkopt(slicer, slicer->cur > slicer->end);
+    bool more = slicer->cur > slicer->end;
```

Comment:
I know this confused me when implementing #16705, but shouldn't it be `not_more` then, when the `cur` pointer is beyond the `end` pointer?

Reply:
(That might be only the problem when `slicer` is used outside the context of this function, but it was not always the value I thought it was, so in #16705 I ended up just using `slicer->end < buf_len` as `more` condition).

Reply:
But in that case you would get an extra block, when `slicer->cur` == `slicer->end` we do not want an extra block so the if should return false.

Reply:
Ah right!